### ISSUE TITLE
Undefined reference 

### ISSUE THREAD

**Issue by JinRhee** [Kin-Zhang/vdbfusion_mapping#5]
Sorry about creating a new issue, I couldn't figure out how to reopen the closed issue.

I'm running into the exact same issue as #3 when trying to catkin_make the workspace after installing glog, open3d and the such.
I can't exactly make out what the solution was other than there being some conflict between open3d and ROS?

(I'm also trying to run this locally rather than using docker)

Any help would be much appreciated!

```shell
/usr/bin/ld: CMakeFiles/vdbfusion_mapping.dir/vdbfusion_mapping_node.cpp.o: in function `main':
vdbfusion_mapping_node.cpp:(.text.startup+0x20f): undefined reference to `ros::init(int&, char**, std::string const&, unsigned int)'
/usr/bin/ld: vdbfusion_mapping_node.cpp:(.text.startup+0x2ac): undefined reference to `ros::NodeHandle::NodeHandle(std::string const&, std::map<std::string, std::string, std::less<std::string>, std::allocator<std::pair<std::string const, std::string> > > const&)'
/usr/bin/ld: vdbfusion_mapping_node.cpp:(.text.startup+0x31e): undefined reference to `ros::NodeHandle::NodeHandle(std::string const&, std::map<std::string, std::string, std::less<std::string>, std::allocator<std::pair<std::string const, std::string> > > const&)'
/usr/bin/ld: /home/user/vdbfusion_mapping/devel/lib/libvdbfusion_ros.so: undefined reference to `open3d::io::WriteTriangleMesh(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, open3d::geometry::TriangleMesh const&, bool, bool, bool, bool, bool, bool)'
collect2: error: ld returned 1 exit status
make[2]: *** [vdbfusion_mapping/vdbfusion_mapping/CMakeFiles/vdbfusion_mapping.dir/build.make:419: /home/user/vdbfusion_mapping/devel/lib/vdbfusion_ros/vdbfusion_mapping] Error 1
make[1]: *** [CMakeFiles/Makefile2:3689: vdbfusion_mapping/vdbfusion_mapping/CMakeFiles/vdbfusion_mapping.dir/all] Error 2

```

**Comment by Kin-Zhang** [comment]
Could you check your environment and Dockerfile, like in Ubuntu 20.04 I fixed v0.16.0 Open3D version here. Let me know after you fixed the open3d version if the problem still exists.

```bash
# Install Open3D dependencies
RUN git clone --depth 1 --branch v0.16.0 https://gitcode.net/mirrors/isl-org/Open3D.git && cd Open3D \
    && util/install_deps_ubuntu.sh && mkdir build && cd build && cmake .. \
    && make -j$(nproc) \
    && make -j$(nproc) all install \
    && cd / \
    && rm -rf /Open3D
```

**Comment by JinRhee** [comment]
Yep, I updated to Open3D v0.18.0 and `catkin_make` worked!

My previously installed Open3D version was v0.13.0, which remained for some reason after following the code above.

Unfortunately, I couldn't access gitcode.net, so I manually uninstalled Open3D and reinstalled the latest version following instructions [here](https://github.com/isl-org/open3d-cmake-find-package) and moved files from `${HOME}/open3d_install` into `/usr/local/` as you mentioned at the bottom of [this file](https://github.com/Kin-Zhang/vdbfusion_mapping/blob/master/assets/readme/install_desktop.md).

Great work and thanks for your help!
